corehq/apps/es/queries.py

```diff
@@ -75,13 +84,31 @@ def search_string_query(search_string, default_fields):
 
     # TODO: add support for searching date ranges.
 
-    return {
+    default_field_query = {
         "query_string": {
             "query": query_string,
             "default_operator": "AND",
             "fields": default_fields,
         }
     }
+    nested_queries = []
+    for nested_field, path in nested_fields:
+        nested_field_query = nested(
+            f"{nested_field}",
+            {
+                "query_string": {
+                    "query": f"{nested_field}.{path}:{query_string}"
+                }
+            }
+        )
+        nested_queries.append(nested_field_query)
+
+    if nested_fields:
+        return BOOL_CLAUSE(
+            SHOULD_CLAUSE([default_field_query] + nested_queries)
```

Comment:
Yes, it does behave the same. I have added a [test](https://github.com/dimagi/commcare-hq/pull/33694/files#diff-ec315c6fbc57958b15ac0681b8f7cd4cd696a18ba451e3387522365e8aa6e57eR164) to test this.

Regarding storing domains for webusers, of course we can do it, but I thought we would try to maintain partiy between ES and couch as much. This info is already captured int the `domain_memberships`, so why duplicate the domain list to another property since we are still able to query well.

Reply:
That test still uses single term queries though - what I mean is that you can currently go to the report and type `testuser34 esoergel` and see a result where the first term matches a username and the second term matches a domain.  What this query would do in that case (I believe) is look for a match for `testuser34 esoergel` across username, first and last name, and phone number, and find no matches, then look for a match in `domain_memberships.domain` and find no matches.

That util is only used to support the admin report, and I don't think we need to worry too much about changing behavior there.  It might also be fine to just drop `domain` from the list of fields checked against, since if you already know the domain, you'd probably just look at that domain's users list.  Though if you're doing a partial match on the domain maybe it would be nice.

Reply:
Ah, got it, I will look into this.